tests/Message/ApplePayAuthorizeRequestTest.php

```diff
@@ -0,0 +1,156 @@
+                'keyCertPath' => $this->keyCertPath,
+                'keyCertPassword' => $this->keyCertPassword,
+                'validationUrl' => $this->validationUrl,
+                'merchantIdentifier' => $this->merchantIdentifier,
+                'displayName' => $this->displayName,
+                'applicationUrl' => $this->applicationUrl
+            )
+        );
+    }
+
+    /**
+     * @return void
+     */
+    public function testGetData()
+    {
+        $data = $this->request->getData();
+        $this->assertSame($this->merchantIdentifier, $data['merchantIdentifier']);
+        $this->assertSame($this->displayName, $data['displayName']);
+        $this->assertSame('web', $data['initiative']);
+        $this->assertSame($this->applicationUrl, $data['initiativeContext']);
+    }
+
+    /**
+     * @return void
+     */
+    public function testValidationUrl()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setValidationUrl($this->validationUrl));
+        $this->assertEquals($this->validationUrl, $request->getValidationUrl());
+    }
+
+    /**
+     * @expectedException        \Omnipay\Common\Exception\InvalidRequestException
+     * @expectedExceptionMessage The validationUrl parameter is required
+     * @return                   void
+     */
+    public function testValidationUrlRequired()
+    {
+        $this->request->setValidationUrl(null);
+        $this->request->getData();
+    }
+
+    /**
+     * @return void
+     */
+    public function testMerchantIdentifier()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setMerchantIdentifier($this->merchantIdentifier));
+        $this->assertSame($this->merchantIdentifier, $request->getMerchantIdentifier());
+    }
+
+    /**
+     * @return void
+     */
+    public function testDisplayName()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setDisplayName($this->displayName));
+        $this->assertSame($this->displayName, $request->getDisplayName());
+    }
+
+    /**
+     * @return void
+     */
+    public function testApplicationUrl()
+    {
+        $request = Mocker::mock('\Omnipay\Vindicia\Message\ApplePayAuthorizeRequest')->makePartial();
+        $request->initialize();
+
+        $this->assertSame($request, $request->setApplicationUrl($this->applicationUrl));
+        $this->assertSame($this->applicationUrl, $request->getApplicationUrl());
+    }
+
+    /**
+     * @return void
+     */
+    public function testSendSuccess()
+    {
+        $this->setMockHttpResponse('ApplePayAuthorizeRequestSuccess.txt');
```

Comment:
what is the reason for using txt instead of xml for mock response?

Reply:
Because the response that we get back is json.

Reply:
ok so should we use .json format for mocking response? if we don't need to, can you add a comment on the response format?

Reply:
Not necessarily, because only the message is json –– the rest of the response is text. I'll add a comment.

Reply:
This is done!

Reply:
ok got it!